component/src/main/java/org/wso2/extension/siddhi/io/hl7/source/RegistrationEventRouting.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+package org.wso2.extension.siddhi.io.hl7.source;
+
+import ca.uhn.hl7v2.protocol.ApplicationRouter;
+
+/**
+ * RegistrationEventRouting is used to route messages to the appropriate application.
+ * This class handles all Message Types and Trigger Events.
+ */
+public class RegistrationEventRouting implements ApplicationRouter.AppRoutingData {
+
+    @Override
+    public String getMessageType() {
+
+        return "*";
```

Comment:
Why we return "*"  here??

Reply:
Hl7 message has several types. It is mentioned to support all the types.
https://www.hl7.org/documentcenter/public_temp_58BF27F2-1C23-BA17-0CCFAA225A24A762/wg/conf/HL7MSH.htm